lib/game_setting/opponent_type.rb

```diff
@@ -0,0 +1,23 @@
+require_relative '../validator'
+require_relative '../messages'
+require_relative './setting_types'
+
+class OpponentType
+    include Messages
+    include Validator
+    include SettingTypes
+    attr_accessor :name, :message
+    
+    def initialize
+        @name = opponent_type
+        @message = opponent_type_msg
+    end
+
+    def valid?(input: nil)
+        opponent_type_valid?(input: input)
+    end
+
+    def clean(input:)
+        input
```

Comment:
Why have this method? What is it cleaning?

Reply:
So clean() is meant to convert the input to whatever the rest of the application expects.
For example, for board size input, it converts it to a integer and passes it back.

Opponent type doesn't do anything, but I didnt know how else to do it if i wanted swappable duck types. I'm not sure if I'm implementing duck types correctly for my game_setting types